local-server.js

```diff
@@ -1,46 +1,52 @@
-var http   = require("http"),
-	url    = require("url"),
-	path   = require("path"),
-	fs     = require("fs"),
-	tachyon= require( './index' ),
+var http = require("http"),
+	url = require("url"),
+	fs = require("fs"),
+	tachyon = require( './index' ),
 	args = process.argv.slice(2),
-	port   = Number( args[0] ) ? args[0] : 8080,
-	debug  = args.indexOf( '--debug' ) > -1
+	port = Number( args[0] ) ? args[0] : 8080,
+	debug = args.indexOf( '--debug' ) > -1;
 
 http.createServer( function( request, response ) {
-	var params = url.parse( request.url, true )
+	var params = url.parse( request.url, true );
 
 	if ( debug ) {
-		console.log( Date(), request.url )
+		console.log( Date(), request.url );
 	}
 
 	try {
-		var imageData = fs.readFileSync( decodeURI( params.pathname.substr(1) ) )
+		var imageData = fs.readFileSync( decodeURI( params.pathname.substr(1) ) );
 	} catch ( err ) {
-		response.writeHead( err.statusCode ? err.statusCode : 500 )
-		response.write( err.message )
-		return response.end()
+		response.writeHead( err.statusCode ? err.statusCode : 500 );
+		response.write( err.message );
+		return response.end();
 	}
 
-	params.query.key = decodeURI( params.pathname.substr(1) )
+	params.query.key = decodeURI( params.pathname.substr(1) );
 
 	return tachyon.resizeBuffer( imageData, params.query, function( err, data, info ) {
-
 		if ( err ) {
+			if ( err.message === 'fallback-to-original' ) {
+				response.writeHead( 200, {
+					'Content-Type': 'image/' + path.extname( params.query.key ).replace( '.', '' ),
```

Comment:
Looks like this is only ever hit if the type is GIF, so can we either hardcode that or pass the type back from resizeBuffer itself? Otherwise potentially leaves open a header injection.

Reply:
yeah I did wonder - I implemented it this way because the error message itself is generic so presumably we might extend it out to other file types in future like APNG.

I'll hard code it for now anyway, then we can consider having a safelist of values in future if a use-case comes up.